src/Microsoft.DotNet.Interactive.Http.Parsing/Parsing/HttpRequestParser.cs

```diff
@@ -482,12 +483,23 @@ private void ConsumeCurrentTokenInto(HttpSyntaxNode node)
                     }
                 }
 
-                if (!((CurrentToken?.Kind is HttpTokenKind.Word or HttpTokenKind.Punctuation) || (IsAtStartOfEmbeddedExpression())))
+                if(CurrentToken?.Kind is HttpTokenKind.NewLine)
                 {
                     break;
                 }
 
-                if (IsAtStartOfEmbeddedExpression())
+                if ((CurrentToken?.Kind is HttpTokenKind.Whitespace))
+                {
+                    var nextSignificantToken = GetNextSignificantToken()?.Text.ToLowerInvariant();
+                    if ((nextSignificantToken is not "http" or "https"))
```

Comment:
This means I can't have a url that contains the word http separated by spaces :) e.g. `https://www.something.com/query?=tell me more about the http protocol`. It may be interesting to check whether REST editor and other extensions allow this.